client/session.go

```diff
@@ -1693,6 +1852,47 @@ func (s *session) streamAndGroupCollectedBlocksMetadata(
 	}
 }
 
+func (s *session) streamAndGroupCollectedBlocksMetadataV2(
+	peersLen int,
+	metadataCh <-chan blocksMetadata,
+	enqueueCh *enqueueChannel,
+) {
+	metadata := make(map[hashAndBlockStart]*receivedBlocks)
+
+	for {
+		m, ok := <-metadataCh
+		if !ok {
+			break
+		}
+
+		key := hashAndBlockStart{
+			hash:       m.id.Hash(),
+			blockStart: m.blocks[0].start.UnixNano(),
+		}
+		received, ok := metadata[key]
+		if !ok || received.submitted {
```

Comment:
If it's already  been submitted we probably don't want to add another entry?  Otherwise we'll submit it twice.

Reply:
This is similarly following the precedent set by the existing code. Is re-submitting it bad? I may have an incomplete understanding here but I thought resubmitting would be an idempotent operation